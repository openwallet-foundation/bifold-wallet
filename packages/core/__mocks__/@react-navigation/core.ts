const navigate = jest.fn()
const dispatch = jest.fn()
const replace = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  replace,
  setOptions: jest.fn(),
  getParent: () => {
    return {
      navigate,
      dispatch,
      replace,
    }
  },
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

const CommonActions = {
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
}

export { useNavigation, useIsFocused, CommonActions }
