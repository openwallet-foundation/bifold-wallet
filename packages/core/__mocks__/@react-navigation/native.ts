const navigate = jest.fn()
const dispatch = jest.fn()
const replace = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  replace,
  setOptions: jest.fn(),
  getParent: () => {
    return {
      navigate,
      dispatch,
      replace,
    }
  },
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

const useRoute = jest.fn().mockReturnValue({ params: {} })

const CommonActions = {
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
}

const useFocusEffect = jest.fn()
const createNavigatorFactory = jest.fn()

export { useNavigation, useIsFocused, useRoute, useFocusEffect, createNavigatorFactory, CommonActions }
