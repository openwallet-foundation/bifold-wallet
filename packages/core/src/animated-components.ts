import ButtonLoading from './components/animated/ButtonLoading'
import ConnectionLoading from './components/animated/ConnectionLoading'
import CredentialAdded from './components/animated/CredentialAdded'
import CredentialPending from './components/animated/CredentialPending'
import LoadingIndicator from './components/animated/LoadingIndicator'
import RecordLoading from './components/animated/RecordLoading'
import SendingProof from './components/animated/SendingProof'
import SentProof from './components/animated/SentProof'

export interface AnimatedComponents {
  ButtonLoading: React.FC
  ConnectionLoading: React.FC
  CredentialAdded: React.FC
  CredentialPending: React.FC
  LoadingIndicator: React.FC
  RecordLoading: React.FC
  SendingProof: React.FC
  SentProof: React.FC
}

export const animatedComponents: AnimatedComponents = {
  ButtonLoading: ButtonLoading,
  ConnectionLoading: ConnectionLoading,
  CredentialAdded: CredentialAdded,
  CredentialPending: CredentialPending,
  LoadingIndicator: LoadingIndicator,
  RecordLoading: RecordLoading,
  SendingProof: SendingProof,
  SentProof: SentProof,
}
