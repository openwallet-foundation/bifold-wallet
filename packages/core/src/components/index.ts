import * as buttons from './buttons'
import * as misc from './misc/index'

export { misc, buttons }
