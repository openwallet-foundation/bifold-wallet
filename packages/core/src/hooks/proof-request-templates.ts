import { ProofRequestTemplate } from '@hyperledger/aries-bifold-verifier'
import { useEffect, useState } from 'react'

import { TOKENS, useServices } from '../container-api'
import { useStore } from '../contexts/store'
import { applyTemplateMarkers, useRemoteProofBundleResolver } from '../utils/proofBundle'

export const useTemplates = (): Array<ProofRequestTemplate> => {
  const [store] = useStore()
  const [proofRequestTemplates, setProofRequestTemplates] = useState<ProofRequestTemplate[]>([])
  const [{ proofTemplateBaseUrl }, logger] = useServices([TOKENS.CONFIG, TOKENS.UTIL_LOGGER])
  const resolver = useRemoteProofBundleResolver(proofTemplateBaseUrl, logger)

  useEffect(() => {
    resolver.resolve(store.preferences.acceptDevCredentials).then((templates) => {
      if (templates) {
        setProofRequestTemplates(applyTemplateMarkers(templates))
      }
    })
  }, [resolver, store.preferences.acceptDevCredentials])

  return proofRequestTemplates
}

export const useTemplate = (templateId: string): ProofRequestTemplate | undefined => {
  const [store] = useStore()
  const [proofRequestTemplate, setProofRequestTemplate] = useState<ProofRequestTemplate | undefined>(undefined)
  const [{ proofTemplateBaseUrl }] = useServices([TOKENS.CONFIG])
  const resolver = useRemoteProofBundleResolver(proofTemplateBaseUrl)

  useEffect(() => {
    resolver.resolveById(templateId, store.preferences.acceptDevCredentials).then((template) => {
      if (template) {
        setProofRequestTemplate(applyTemplateMarkers(template))
      }
    })
  }, [resolver, templateId, store.preferences.acceptDevCredentials])

  return proofRequestTemplate
}
