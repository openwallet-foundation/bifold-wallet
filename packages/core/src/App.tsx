import React, { useEffect } from 'react'
import { StatusBar } from 'react-native'
import SplashScreen from 'react-native-splash-screen'
import Toast from 'react-native-toast-message'

import { useNavigationContainerRef } from '@react-navigation/native'
import { isTablet } from 'react-native-device-info'
import Orientation from 'react-native-orientation-locker'
import { animatedComponents } from './animated-components'
import ErrorModal from './components/modals/ErrorModal'
import toastConfig from './components/toast/ToastConfig'
import { tours } from './constants'
import { Container, ContainerProvider } from './container-api'
import { AnimatedComponentsProvider } from './contexts/animated-components'
import { AuthProvider } from './contexts/auth'
import NavContainer from './contexts/navigation'
import { NetworkProvider } from './contexts/network'
import { StoreProvider } from './contexts/store'
import { ThemeProvider } from './contexts/theme'
import { TourProvider } from './contexts/tour/tour-provider'
import { initStoredLanguage } from './localization'
import RootStack from './navigators/RootStack'
import { bifoldTheme, themes } from './theme'

const createApp = (container: Container): React.FC => {
  const AppComponent: React.FC = () => {
    const navigationRef = useNavigationContainerRef()

    useEffect(() => {
      initStoredLanguage().then()
    }, [])

    useEffect(() => {
      // Hide the native splash / loading screen so that our
      // RN version can be displayed.
      SplashScreen.hide()
    }, [])

    if (!isTablet()) {
      Orientation.lockToPortrait()
    }

    return (
      <ContainerProvider value={container}>
        <StoreProvider>
          <ThemeProvider themes={themes} defaultThemeName={bifoldTheme.themeName}>
            <NavContainer navigationRef={navigationRef}>
              <AnimatedComponentsProvider value={animatedComponents}>
                <AuthProvider>
                  <NetworkProvider>
                    <StatusBar
                      hidden={false}
                      barStyle="light-content"
                      backgroundColor={bifoldTheme.ColorPalette.brand.primary}
                      translucent={false}
                    />
                    <ErrorModal />
                    <TourProvider tours={tours} overlayColor={'gray'} overlayOpacity={0.7}>
                      <RootStack />
                    </TourProvider>
                    <Toast topOffset={15} config={toastConfig} />
                  </NetworkProvider>
                </AuthProvider>
              </AnimatedComponentsProvider>
            </NavContainer>
          </ThemeProvider>
        </StoreProvider>
      </ContainerProvider>
    )
  }

  return AppComponent
}

export default createApp
