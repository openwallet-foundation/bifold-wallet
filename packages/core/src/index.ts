/* eslint-disable import/no-cycle */
import type { OnboardingStyleSheet } from './screens/Onboarding'

import { Agent } from '@credo-ts/core'
import AgentProvider from '@credo-ts/react-hooks'

import createApp from './App'
import * as components from './components'
import { ButtonImpl as Button, ButtonType, Button as IButton } from './components/buttons/Button'
import IconButton, { ButtonLocation } from './components/buttons/IconButton'
import BulletPoint from './components/inputs/BulletPoint'
import CheckBoxRow from './components/inputs/CheckBoxRow'
import LimitedTextInput from './components/inputs/LimitedTextInput'
import NotificationListItem from './components/listItems/NotificationListItem'
import ContentGradient from './components/misc/ContentGradient'
import CredentialCard from './components/misc/CredentialCard'
import ErrorBoundaryWrapper from './components/misc/ErrorBoundary'
import FauxHeader from './components/misc/FauxHeader'
import InfoBox, { InfoBoxType } from './components/misc/InfoBox'
import QRScannerTorch from './components/misc/QRScannerTorch'
import DeveloperModal from './components/modals/DeveloperModal'
import ErrorModal from './components/modals/ErrorModal'
import SafeAreaModal from './components/modals/SafeAreaModal'
import Record from './components/record/Record'
import InfoTextBox from './components/texts/InfoTextBox'
import Link from './components/texts/Link'
import Text from './components/texts/Text'
import { ThemedText } from './components/texts/ThemedText'
import { ToastType } from './components/toast/BaseToast'
import toastConfig from './components/toast/ToastConfig'
import { AttachTourStep } from './components/tour/AttachTourStep'
import { credentialOfferTourSteps } from './components/tour/CredentialOfferTourSteps'
import { credentialsTourSteps } from './components/tour/CredentialsTourSteps'
import { homeTourSteps } from './components/tour/HomeTourSteps'
import { proofRequestTourSteps } from './components/tour/ProofRequestTourSteps'
import { TourBox } from './components/tour/TourBox'
import { Banner, BannerMessage, BannerSection } from './components/views/Banner'
import HomeFooterView from './components/views/HomeFooterView'
import KeyboardView from './components/views/KeyboardView'
import { attemptLockoutConfig, PINRules, tours, walletTimeout } from './constants'
import { defaultConfig, defaultHistoryEventsLogger } from './container-impl'
import * as contexts from './contexts'
import { ActivityProvider, AutoLockTime, useActivity } from './contexts/activity'
import { AuthProvider } from './contexts/auth'
import NavContainer from './contexts/navigation'
import { NetworkProvider } from './contexts/network'
import { useTour } from './contexts/tour/tour-context'
import { TourProvider } from './contexts/tour/tour-provider'
import { useDeveloperMode } from './hooks/developer-mode'
import usePreventScreenCapture from './hooks/screen-capture'
import useBifoldAgentSetup from './hooks/useBifoldAgentSetup'
import { OpenIDCredentialRecordProvider } from './modules/openid/context/OpenIDCredentialRecordProvider'
import { DefaultScreenLayoutOptions } from './navigators/defaultLayoutOptions'
import { DefaultScreenOptionsDictionary, useDefaultStackOptions } from './navigators/defaultStackOptions'
import AttemptLockout from './screens/AttemptLockout'
import Biometry from './screens/Biometry'
import Developer from './screens/Developer'
import Onboarding from './screens/Onboarding'
import OnboardingPages from './screens/OnboardingPages'
import Preface from './screens/Preface'
import Scan from './screens/Scan'
import Splash from './screens/Splash'
import Terms from './screens/Terms'
import UpdateAvailable from './screens/UpdateAvailable'
import { bifoldLoggerInstance } from './services/bifoldLogger'
import { isBiometricsActive, loadLoginAttempt } from './services/keychain'
import { BifoldLogger } from './services/logger'
import { AbstractBifoldLogger } from './services/AbstractBifoldLogger'
import { DeepPartial, ThemeBuilder } from './theme-builder'
import * as types from './types'
import { CredentialListFooterProps } from './types/credential-list-footer'

export { animatedComponents } from './animated-components'
export { EventTypes, LocalStorageKeys } from './constants'
export { AnimatedComponentsProvider, useAnimatedComponents } from './contexts/animated-components'
export { useAuth } from './contexts/auth'
export { useNetwork } from './contexts/network'
export { DispatchAction, reducer, default as Store } from './contexts/reducers/store'
export { defaultState, mergeReducers, StoreContext, StoreProvider, useStore } from './contexts/store'
export { ThemeProvider, useTheme } from './contexts/theme'
export { useDeepLinks } from './hooks/deep-links'
export { initLanguages, initStoredLanguage, Locales, translationResources } from './localization'
export * from './navigators'
export { createStyles } from './screens/OnboardingPages'
export * from './services/storage'
export { bifoldTheme, ColorPalette, Assets as ImageAssets } from './theme'
export * from './types/attestation'
export { BifoldError } from './types/error'
export { Screens, Stacks, TabStacks } from './types/navigators'
export * from './types/version-check'
export { createLinkSecretIfRequired, getAgentModules } from './utils/agent'
export { getCredentialIdentifiers, isValidAnonCredsCredential } from './utils/credential'
export {
  connectFromScanOrDeepLink,
  formatTime,
  getConnectionName,
  removeExistingInvitationsById,
  useCredentialConnectionLabel,
} from './utils/helpers'
export { getIndyLedgers, IndyLedger, readIndyLedgersFromFile, writeIndyLedgersToFile } from './utils/ledger'
export { statusBarStyleForColor, StatusBarStyles } from './utils/luminance'
export { migrateToAskar } from './utils/migration'
export { buildFieldsFromAnonCredsCredential } from './utils/oca'
export { testIdForAccessabilityLabel, testIdWithKey } from './utils/testable'

export type { AnimatedComponents } from './animated-components'
export type { ReducerAction } from './contexts/reducers/store'
export type { RenderProps, TourStep } from './contexts/tour/tour-context'
export type {
  IAssets,
  IBrandColors,
  IColorPalette,
  IErrorColors,
  IFontAttributes,
  IGrayscaleColors,
  IInlineInputMessage,
  IInputAttributes,
  IInputs,
  INotificationColors,
  ISemanticColors,
  ISpacing,
  ISVGAssets,
  ITabTheme,
  ITextTheme,
  ITheme,
} from './theme'
export type { GenericFn } from './types/fn'
export { BasicMessageMetadata, CredentialMetadata } from './types/metadata'
export type { basicMessageCustomMetadata, credentialCustomMetadata } from './types/metadata'
export type { ContactStackParams, NotificationStackParams, OnboardingStackParams } from './types/navigators'
export type { WalletSecret } from './types/security'
export type {
  LoginAttempt as LoginAttemptState,
  Migration as MigrationState,
  Onboarding as OnboardingState,
  PersistentState,
  Preferences as PreferencesState,
  State,
  Tours as ToursState,
  VersionInfo,
} from './types/state'
export type { BifoldAgent } from './utils/agent'
export type { IndyLedgerConfig, IndyLedgerFileSystem, IndyLedgerJSON, IndyLedgersRecord } from './utils/ledger'
export type { OnboardingStyleSheet }

export type { InlineMessageProps } from './components/inputs/InlineErrorText'

export { InlineErrorPosition } from './types/error'

export * from './container-api'
export { MainContainer } from './container-impl'
export type { CredentialListFooterProps }

export type { Config, HistoryEventsLoggerConfig } from './types/config'
export type { OnboardingTask, ScreenLayoutConfig } from './types/navigators'

export { LockoutReason } from './contexts/auth'
export type { OnboardingStackProps } from './navigators/OnboardingStack'
export type { SplashProps } from './screens/Splash'
export { BaseTourID } from './types/tour'

export type { BannerSectionProps } from './components/views/Banner'

export {
  ActivityProvider,
  Agent,
  AgentProvider,
  AttachTourStep,
  AttemptLockout,
  attemptLockoutConfig,
  AuthProvider,
  AutoLockTime,
  Banner,
  BannerSection,
  AbstractBifoldLogger,
  BifoldLogger,
  bifoldLoggerInstance,
  Biometry,
  BulletPoint,
  Button,
  ButtonLocation,
  ButtonType,
  CheckBoxRow,
  components,
  ContentGradient,
  contexts,
  createApp,
  CredentialCard,
  credentialOfferTourSteps,
  credentialsTourSteps,
  defaultConfig,
  defaultHistoryEventsLogger,
  DefaultScreenLayoutOptions,
  DefaultScreenOptionsDictionary,
  Developer,
  DeveloperModal,
  ErrorBoundaryWrapper,
  ErrorModal,
  FauxHeader,
  HomeFooterView as HomeContentView,
  homeTourSteps,
  IconButton,
  InfoBox,
  InfoBoxType,
  InfoTextBox,
  isBiometricsActive,
  KeyboardView,
  LimitedTextInput,
  Link,
  loadLoginAttempt,
  NavContainer,
  NetworkProvider,
  NotificationListItem,
  Onboarding,
  OnboardingPages,
  OpenIDCredentialRecordProvider,
  PINRules,
  Preface,
  proofRequestTourSteps,
  QRScannerTorch,
  Record,
  SafeAreaModal,
  Scan,
  Splash,
  Terms,
  Text,
  ThemeBuilder,
  ThemedText,
  toastConfig,
  ToastType,
  TourBox,
  TourProvider,
  tours,
  types,
  UpdateAvailable,
  useActivity,
  useBifoldAgentSetup,
  useDefaultStackOptions,
  useDeveloperMode,
  usePreventScreenCapture,
  useTour,
  walletTimeout,
}
export type { BannerMessage, DeepPartial, IButton }
