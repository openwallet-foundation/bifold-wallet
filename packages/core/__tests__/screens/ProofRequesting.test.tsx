import { INDY_PROOF_REQUEST_ATTACHMENT_ID, V1RequestPresentationMessage } from '@credo-ts/anoncreds'
import {
  ConnectionRecord,
  DidExchangeRole,
  DidExchangeState,
  OutOfBandInvitation,
  ProofExchangeRecord,
  ProofRole,
  ProofState,
} from '@credo-ts/core'
import { Attachment, AttachmentData } from '@credo-ts/core/build/decorators/attachment/Attachment'
import { useConnections } from '@credo-ts/react-hooks'
import * as verifier from '@hyperledger/aries-bifold-verifier'
import { getProofRequestTemplates } from '@hyperledger/aries-bifold-verifier'
import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock'
import { act, cleanup, fireEvent, render, waitFor } from '@testing-library/react-native'
import React from 'react'

import { useNavigation as testUseNavigation } from '../../__mocks__/@react-navigation/native'
import { testIdWithKey } from '../../src'
import * as proofRequestTemplatesHooks from '../../src/hooks/proof-request-templates'
import ProofRequesting from '../../src/screens/ProofRequesting'
import { BasicAppContext } from '../helpers/app'

jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo)
jest.mock('@hyperledger/aries-bifold-verifier', () => {
  const original = jest.requireActual('@hyperledger/aries-bifold-verifier')
  return {
    ...original,
    __esModule: true,
    createConnectionlessProofRequestInvitation: jest.fn(original.createConnectionlessProofRequestInvitation),
  }
})
jest.mock('react-native-device-info', () => {
  return require('../../__mocks__/custom/react-native-device-info')
})

jest.useFakeTimers({ legacyFakeTimers: true })
jest.spyOn(global, 'setTimeout')

const templates = getProofRequestTemplates(true)
const template = templates[0]
const templateId = template.id

const proof_request = {
  name: 'proof-request',
  version: '1.0',
  nonce: '1298236324864',
  requested_attributes: {
    attribute_1: {
      names: ['first_name', 'second_name'],
    },
  },
}

const requestPresentationMessage = new V1RequestPresentationMessage({
  comment: 'some comment',
  requestAttachments: [
    new Attachment({
      id: INDY_PROOF_REQUEST_ATTACHMENT_ID,
      mimeType: 'application/json',
      data: new AttachmentData({
        json: proof_request,
      }),
    }),
  ],
})
const proofRecord = new ProofExchangeRecord({
  connectionId: undefined,
  threadId: requestPresentationMessage.id,
  state: ProofState.RequestSent,
  protocolVersion: 'V1',
  isVerified: true,
  role: ProofRole.Prover,
})
const invitation = new OutOfBandInvitation({
  id: 'test',
  label: 'Test Invitation',
  services: [],
})
const invitationUrl = 'http://example.com/ssi?oob=eyJ'

const data = {
  request: requestPresentationMessage,
  proofRecord,
  invitation,
  invitationUrl,
}

describe('ProofRequesting Component', () => {
  const testContactRecord1 = new ConnectionRecord({
    id: '1',
    did: '9gtPKWtaUKxJir5YG2VPxX',
    theirLabel: 'Faber',
    role: DidExchangeRole.Responder,
    theirDid: '2SBuq9fpLT8qUiQKr2RgBe',
    threadId: '1',
    state: DidExchangeState.Completed,
    createdAt: new Date('2020-01-01T00:00:00.000Z'),
  })
  const testContactRecord2 = new ConnectionRecord({
    id: '2',
    did: '2SBuq9fpLT8qUiQKr2RgBe',
    role: DidExchangeRole.Requester,
    theirLabel: 'Bob',
    theirDid: '9gtPKWtaUKxJir5YG2VPxX',
    threadId: '1',
    state: DidExchangeState.Completed,
    createdAt: new Date('2020-01-01T00:00:00.000Z'),
  })
  afterEach(() => {
    cleanup()
  })
  beforeEach(() => {
    jest.clearAllMocks()

    // @ts-expect-error useConnections will be replaced with a mock which does have this method
    useConnections.mockReturnValue({ records: [testContactRecord1, testContactRecord2] })
    jest.spyOn(verifier, 'createConnectionlessProofRequestInvitation').mockReturnValue(Promise.resolve(data))
    jest.spyOn(proofRequestTemplatesHooks, 'useTemplate').mockReturnValue(template)
  })

  const renderView = (params?: { templateId: string; predicateValues: any }) => {
    return render(
      <BasicAppContext>
        <ProofRequesting navigation={testUseNavigation() as any} route={{ params } as any} />
      </BasicAppContext>
    )
  }

  test('renders correctly', async () => {
    const tree = renderView({ templateId, predicateValues: {} })

    await waitFor(
      () => {
        expect(tree.getByTestId(testIdWithKey('LoadingActivityIndicator'))).toBeDefined()
        expect(tree).toMatchSnapshot()
      },
      { timeout: 50000 }
    )

    await act(async () => {})
    expect(tree).toMatchSnapshot()
  })

  test('generate new qr works correctly', async () => {
    const tree = renderView({ templateId, predicateValues: {} })

    const generateQRButton = tree.getByTestId(testIdWithKey('GenerateNewQR'))
    expect(generateQRButton).not.toBeNull()
    fireEvent(generateQRButton, 'press')

    await waitFor(
      () => {
        expect(tree.getByTestId(testIdWithKey('LoadingActivityIndicator'))).toBeDefined()
        expect(tree).toMatchSnapshot()
      },
      { timeout: 50000 }
    )
  })
})
