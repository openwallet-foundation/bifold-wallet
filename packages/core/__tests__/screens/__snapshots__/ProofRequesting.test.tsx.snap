// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProofRequesting Component generate new qr works correctly 1`] = `
<RNCSafeAreaView
  edges={
    Object {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "off",
    }
  }
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexGrow": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 730,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "marginTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
          testID="com.ariesbifold:id/LoadingActivityIndicator"
        >
          <Image
            source={
              Object {
                "default": "",
              }
            }
            style={
              Object {
                "height": "33%",
                "objectFit": "contain",
                "width": "33%",
              }
            }
            testID="com.ariesbifold:id/LoadingActivityIndicatorImage"
          />
          <View
            collapsable={false}
            style={
              Object {
                "position": "absolute",
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <
              fill="#FFFFFF"
              height={200}
              width={200}
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 20,
            "paddingVertical": 16,
            "textAlign": "center",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Object {
                "color": "#000000",
                "marginTop": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Verifier.ScanQR
        </Text>
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 26,
                "fontWeight": "bold",
              },
              Object {
                "marginTop": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          Student full name
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "marginBottom": 10,
        "marginHorizontal": 20,
        "marginTop": "auto",
      }
    }
  >
    <View
      accessibilityLabel="Verifier.RefreshQR"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(53, 130, 63, 0.35)",
          "borderRadius": 4,
          "opacity": 1,
          "padding": 16,
        }
      }
      testID="com.ariesbifold:id/GenerateNewQR"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
                false,
              ],
            ]
          }
        >
          Verifier.RefreshQR
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`ProofRequesting Component renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Object {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "off",
    }
  }
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexGrow": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 730,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "marginTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
          testID="com.ariesbifold:id/LoadingActivityIndicator"
        >
          <Image
            source={
              Object {
                "default": "",
              }
            }
            style={
              Object {
                "height": "33%",
                "objectFit": "contain",
                "width": "33%",
              }
            }
            testID="com.ariesbifold:id/LoadingActivityIndicatorImage"
          />
          <View
            collapsable={false}
            style={
              Object {
                "position": "absolute",
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <
              fill="#FFFFFF"
              height={200}
              width={200}
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 20,
            "paddingVertical": 16,
            "textAlign": "center",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Object {
                "color": "#000000",
                "marginTop": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Verifier.ScanQR
        </Text>
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 26,
                "fontWeight": "bold",
              },
              Object {
                "marginTop": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          Student full name
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "marginBottom": 10,
        "marginHorizontal": 20,
        "marginTop": "auto",
      }
    }
  >
    <View
      accessibilityLabel="Verifier.RefreshQR"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(53, 130, 63, 0.35)",
          "borderRadius": 4,
          "opacity": 1,
          "padding": 16,
        }
      }
      testID="com.ariesbifold:id/GenerateNewQR"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
                false,
              ],
            ]
          }
        >
          Verifier.RefreshQR
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`ProofRequesting Component renders correctly 2`] = `
<RNCSafeAreaView
  edges={
    Object {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "off",
    }
  }
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexGrow": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 730,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "marginTop": 15,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 20,
            "paddingVertical": 16,
            "textAlign": "center",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Object {
                "color": "#000000",
                "marginTop": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Verifier.ScanQR
        </Text>
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 26,
                "fontWeight": "bold",
              },
              Object {
                "marginTop": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          Student full name
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "marginBottom": 10,
        "marginHorizontal": 20,
        "marginTop": "auto",
      }
    }
  >
    <View
      accessibilityLabel="Verifier.RefreshQR"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#42803E",
          "borderRadius": 4,
          "opacity": 1,
          "padding": 16,
        }
      }
      testID="com.ariesbifold:id/GenerateNewQR"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
                false,
                false,
              ],
            ]
          }
        >
          Verifier.RefreshQR
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
